OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/alias/pointsto/AbstractPointsToBasedAliasAnalysis.scala

```diff
@@ -0,0 +1,302 @@
+            .foreach(pts => {
+
+                if (pts.isEPK) {
+                    state.addDependency(pts)
+                    state.addFieldDependency(field, pts)
+                } else {
+
+                    val fieldReferenceEntity = (field.fieldReference, pts.e)
+                    pts.ub.forNewestNElements(pts.ub.numElements - state.pointsToElementsHandled(
+                        field,
+                        fieldReferenceEntity
+                    )) {
+                        value =>
+                            {
+                                allocationSites += value
+                                state.incPointsToElementsHandled(field, fieldReferenceEntity)
+                            }
+                    }
+                }
+            })
+
+        allocationSites.map(allocSite =>
+            propertyStore((allocSite, declaredFields(field.fieldReference.field)), pointsToPropertyKey)
+        )
+    }
+
+    /**
+     * Retrieves the points-to set of the return value of the given method.
+     */
+    private[this] def getPointsToOfReturnValue(callContext: Context): EOptionP[Entity, PointsToSet] = {
+        propertyStore(
+            callContext,
+            pointsToPropertyKey
+        )
+    }
+
+    /**
+     * Handles all given points-to entities associated with given [[AliasSourceElement]] by updating the analysis state.
+     */
+    private[this] def handlePointsToEntities(ase: AliasSourceElement, eps: Iterable[EOptionP[Entity, PointsToSet]])(
+        implicit
+        state:   AnalysisState,
+        context: AnalysisContext
+    ): Unit = {
+        eps.foreach(handlePointsToEntity(ase, _))
+    }
+
+    /**
+     * Handles the given points-to entity associated with the given [[AliasSourceElement]] by updating the analysis state.
+     */
+    private[this] def handlePointsToEntity(ase: AliasSourceElement, eps: EOptionP[Entity, PointsToSet])(
+        implicit
+        state:   AnalysisState,
+        context: AnalysisContext
+    ): Unit = {
+
+        val pointsToEntity: Entity = eps.e
+
+        if (eps.isEPK) {
+            state.addDependency(eps)
+            state.addElementDependency(ase, eps)
+        } else handlePointsToSet(ase, pointsToEntity, eps.ub)
+    }
+
+    /**
+     * Handles the given points-to set associated with the given [[AliasSourceElement]] by updating the analysis state.
+     *
+     * @param ase The [[AliasSourceElement]] the given pointsTo set is associated with.
+     * @param pointsToEntity The pointsTo entity the pointTo Set belongs to.
+     * @param pointsToSet The pointsTo set to handle.
+     */
+    private[this] def handlePointsToSet(ase: AliasSourceElement, pointsToEntity: Entity, pointsToSet: PointsToSet)(
+        implicit
+        state:   AnalysisState,
+        context: AnalysisContext
+    ): Unit = {
+        pointsToSet.forNewestNElements(pointsToSet.numElements - state.pointsToElementsHandled(ase, pointsToEntity)) {
```

Comment:
Why use a stored value of handled elements when you can just store the old points-to set and use its numElements?